sound/soc/sof/topology.c

```diff
@@ -2000,9 +2001,16 @@ static int sof_set_dai_config(struct snd_sof_dev *sdev, u32 size,
 			dai->dai_config = kmemdup(config, size, GFP_KERNEL);
 			if (!dai->dai_config)
 				return -ENOMEM;
+
+			found = 1;
 		}
 	}
 
+	if (!found) {
+		dev_warn(sdev->dev, "warning: failed to find dai for dai link %s",
+			 link->name);
```

Comment:
Should we really be returning 0 in that case?

Reply:
Yes, I will add a comment for it.  I discussed with @ranj063 that dai may not be supported by topology although the dai link is defined in machine driver, so none of dai would be found. But we should print a warning to notify user: is this really you want?

Reply:
@plbossart the rationale for returning a 0 is that even if we dont find a dai to set the config for, there will really be no negative consequences of having just a BE dai config without a pipeline that uses it. 
Of course, logically it probably makes no sense. Thats why the warning here.